File: pset1/greedy.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>

int main(void)
{
    
    int count = 0;
    printf("How much change is owed?\n");
    float change = GetFloat();
    
    // make sure the input is not a negtive number
    while(change < 0 ) 
    {
        printf("How much change is owed?\n");
        change = GetFloat();
    }
        // change float -> int 
        // cent -> $
        int amount =  round(change * 100);
    // the algorithm
    while(amount > 0)
    {
        if((amount - 25) >= 0 ){
            amount = amount - 25;
            count++;
        }
        else if((amount - 10) >= 0 ){
            amount = amount - 10 ;
            count++;
        }
         else if((amount - 5) >= 0 ){
            amount = amount - 5 ;
            count++;
        }
         else if((amount - 1) >= 0 ){
            amount = amount - 1 ;
            count++;
        }
    }
    printf("%d\n", count);
}

File: pset6/greedy.py
import sys

count = 0
change = float(input("How much change is owed?\n"))
while True:
    if change > 0:
        break
    change = float(input("How much change is owed?\n"))
amount = round(change * 100)
while amount > 0:
    if (amount-25)>=0:
        amount-=25
        count+=1
    elif (amount-10)>=0:
        amount-=10
        count+=1
    elif (amount-5)>=0:
        amount-=5
        count+=1
    elif(amount-1)>=0:
        amount-=1
        count+=1
print(count)        